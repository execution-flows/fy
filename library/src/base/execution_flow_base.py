# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
#  file, You can obtain one at https://mozilla.org/MPL/2.0/.

import abc
from typing import Generic, TypeVar

ResultT = TypeVar("ResultT")


class ExecutionFlowBase(Generic[ResultT], abc.ABC):
    @abc.abstractmethod
    def __call__(self) -> ResultT:
        raise NotImplementedError()
