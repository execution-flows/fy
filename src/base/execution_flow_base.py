import abc
from typing import Generic, TypeVar

ResultT = TypeVar("ResultT")


class ExecutionFlowBase(Generic[ResultT], abc.ABC):
    @abc.abstractmethod
    def __call__(self) -> ResultT:
        raise NotImplementedError()
