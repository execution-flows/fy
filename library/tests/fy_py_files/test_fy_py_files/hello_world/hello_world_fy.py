"""fy
flow HelloWorld -> None:
"""

from fy_core.base.flow_base import FlowBase


# fy:start ===>>>
class HelloWorld_Flow(
    # Base
    FlowBase[None],
):
    def __call__(self) -> None:
        # fy:end <<<===
        pass
