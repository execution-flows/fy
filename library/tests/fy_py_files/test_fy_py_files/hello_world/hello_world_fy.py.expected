"""fy
flow HelloWorld -> None:
"""

from base.flow_base import FlowBase


# fy:start <<<===
class HelloWorld_Flow(
    # Base
    FlowBase[None],
):
    def __call__(self) -> None:
        # fy:end <<<===
        pass
